spec/acceptance/service_brokers_v3_spec.rb

```diff
@@ -0,0 +1,261 @@
+            expect(parsed_body.fetch('resources').first.fetch('name')).to eq('test-broker-1')
+            expect(parsed_body.fetch('resources').last.fetch('name')).to eq('test-broker-2')
+          end
+        end
+
+        context 'descending order' do
+          before(:each) do
+            get('/v3/service_brokers?order_by=-name', {}, admin_headers)
+          end
+
+          it 'returns 200 OK and a body containg the brokers ordered by created at time' do
+            expect(last_response).to have_status_code(200)
+
+            parsed_body = JSON.parse(last_response.body)
+
+            expect(parsed_body.fetch('resources').length).to eq(2)
+            expect(parsed_body.fetch('resources').first.fetch('name')).to eq('test-broker-2')
+            expect(parsed_body.fetch('resources').last.fetch('name')).to eq('test-broker-1')
+          end
+        end
+      end
+
+      context 'when requesting with a space guid filter' do
+        before(:each) do
+          get("/v3/service_brokers?space_guids=#{space.guid}", {}, admin_headers)
+        end
+
+        it 'returns 200 OK and a body containing one broker matching the space guid filter' do
+          expect(last_response).to have_status_code(200)
+
+          parsed_body = JSON.parse(last_response.body)
+
+          expect(parsed_body.fetch('resources').length).to eq(1)
+          expect(parsed_body.fetch('resources').first.fetch('name')).to eq('test-broker-2')
+        end
+      end
+
+      context 'when requesting with a space guid filter for a random space guid' do
+        before(:each) do
+          get('/v3/service_brokers?space_guids=random-space-guid', {}, admin_headers)
+        end
+
+        it 'returns 200 OK and a body containing no broker' do
+          expect(last_response).to have_status_code(200)
+
+          parsed_body = JSON.parse(last_response.body)
+
+          expect(parsed_body.fetch('resources').length).to eq(0)
+        end
+      end
+
+      context 'when requesting with a names filter' do
+        before(:each) do
+          get("/v3/service_brokers?names=#{service_broker1.name}", {}, admin_headers)
+        end
+
+        it 'returns 200 OK and a body containing one broker matching the names filter' do
+          expect(last_response).to have_status_code(200)
+
+          parsed_body = JSON.parse(last_response.body)
+
+          expect(parsed_body.fetch('resources').length).to eq(1)
+          expect(parsed_body.fetch('resources').first.fetch('name')).to eq(service_broker1.name)
+        end
+      end
+    end
+
+    context 'fetching brokers as a non-admin user' do
+      let(:user) { VCAP::CloudController::User.make }
+      let(:org) { VCAP::CloudController::Organization.make }
+      let(:space) { VCAP::CloudController::Space.make(organization: org) }
+      let!(:object) { VCAP::CloudController::ServiceBroker.make }
+      let!(:broker_with_space) { VCAP::CloudController::ServiceBroker.make space: space }
```

Comment:
We think here we need another service-broker object that is not space-scoped, to actually properly test that space developer and other non-admin users can't see it.

Reply:
Agree.

Reply:
done